### PR TITLE
chore: remove default value from meshconfig CRD

### PR DESCRIPTION
**Pull request by allenlsy**
Signed-off-by: Allen Leigh <allenlsy@gmail.com>

**Description**:

Remove default value settings from MeshConfig CRD. Default values should be defined in `values.yaml`

Addressed issue #3908 

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [X] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

**Testing done**:

CI test

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

No

2. Is this a breaking change?

No

### CODE REVIEW

File: charts/osm/values.yaml [hunk 1]
```diff
@@ -188,7 +188,7 @@ OpenServiceMesh:
     # -- Toggles Envoy's tracing functionality on/off for all sidecar proxies in the mesh
     enable: false
     # -- Address of the tracing collector service (must contain the namespace). When left empty, this is computed in helper template to "jaeger.<osm-namespace>.svc.cluster.local". Please override for BYO-tracing as documented in tracing.md
```
Comment (jaellio) [inline charts/osm/values.yaml:190]: The helper template computes `osm.tracingAddress` as mentioned in the comment. However, I don't believe `osm.tracingAddress` is referenced in any template. Only `.Values.OpenServiceMesh.tracing.address` is referenced in the `preset-mesh-config.yaml`. The above comment could be updated and the helper function removed, or the `preset-mesh-config.yaml` could be updated to reference `osm.tracingAddress`.

**Review by jaellio — APPROVED** [review]

File: charts/osm/README.md [hunk 2]
```diff
@@ -139,7 +139,7 @@ The following table lists the configurable parameters of the osm chart and their
 | OpenServiceMesh.prometheus.retention.time | string | `"15d"` | Prometheus data retention time |
 | OpenServiceMesh.pspEnabled | bool | `false` | Run OSM with PodSecurityPolicy configured |
 | OpenServiceMesh.sidecarImage | string | `"envoyproxy/envoy-alpine:v1.18.3"` | Envoy sidecar image |
-| OpenServiceMesh.tracing.address | string | `""` | Address of the tracing collector service (must contain the namespace). When left empty, this is computed in helper template to "jaeger.<osm-namespace>.svc.cluster.local". Please override for BYO-tracing as documented in tracing.md |
+| OpenServiceMesh.tracing.address | string | `"jaeger.osm-system.svc.cluster.local"` | Address of the tracing collector service (must contain the namespace). When left empty, this is computed in helper template to "jaeger.<osm-namespace>.svc.cluster.local". Please override for BYO-tracing as documented in tracing.md |
```
[the remark this answers is not in the document]
Comment (SanyaKochhar) [reply charts/osm/README.md:142]: https://docs.openservicemesh.io/docs/guides/observability/tracing/

File: charts/osm/values.yaml (hunk 1 above)
Comment (SanyaKochhar) [reply charts/osm/values.yaml:190]: hmm so previously the computed value from
https://github.com/openservicemesh/osm/blob/main/charts/osm/templates/_helpers.tpl#L6-L10 used to be referenced in osm-configmap.yaml as added with this change: https://github.com/openservicemesh/osm/commit/13ef51563fe9d7b8a36ad3547f96244da8f7faa6

I'm not sure how the osm-config values are set so unsure where/if this is used anymore

File: charts/osm/values.yaml [hunk 3]
```diff
@@ -188,7 +188,7 @@ OpenServiceMesh:
...
-    address: ""
+    address: "jaeger.osm-system.svc.cluster.local"
```
[the remark this answers is not in the document]
Comment (allenlsy) [reply charts/osm/values.yaml:191]: This is just the default. Use can pass in customized value via install command.

**Review by shashankram — APPROVED** [review]

File: charts/osm/values.yaml (hunk 1 above)
Comment (allenlsy) [reply charts/osm/values.yaml:190]: osm-config was the ConfigMap format of OSM configuration. It was removed in v0.9.0

**Pull request merged**